libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -182,10 +177,7 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
 
     _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_INLINE_VISIBILITY
     arrival_token arrive(_CUDA_VSTD::ptrdiff_t __update = 1) {
-#if (_LIBCUDACXX_DEBUG_LEVEL >= 2)
-        _LIBCUDACXX_DEBUG_ASSERT(__update >= 0);
-        _LIBCUDACXX_DEBUG_ASSERT(__expected_unit >=0);
-#endif
+        _LIBCUDACXX_DEBUG_ASSERT(__update >= 0, "Arrival count update must be non-negative.");
```

Comment:
Why was the check for expected unit dropped?

Reply:
It is a static variable defined in the base 
```cpp
    static constexpr uint64_t __expected_unit = 1ull;
```

Asserting it at runtime does not add value. We could add a static assert if you really want

Also it is not used in the function at all, so I do not see any reason to check for it here